metrics/network/cpu_statistics_iperf.sh

```diff
@@ -0,0 +1,108 @@
+# results that help to detect performace regressions compared
+# with the actual throughput.
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Test name
+TEST_NAME="${TEST_NAME:-cpu information}"
+# Image name
+IMAGE="${IMAGE:-local-iperf}"
+# Dockerfile
+DOCKERFILE="${SCRIPT_PATH}/iperf_dockerfile/Dockerfile"
+# Client name
+CLIENT_NAME="${CLIENT_NAME:-client_iperf}"
+# Server name
+SERVER_NAME="${SERVER_NAME:-server_iperf}"
+# Target bandwidth
+BANDWIDTH="${BANDWIDTH:-1G}"
+# Time in seconds to transmit
+TIMEOUT="${TIMEOUT:-160}"
+# Sample period in seconds to measure
+SAMPLE_PERIOD="${SAMPLE_PERIOD:-120}"
+# File to save perf results
+TMP_FILE=$(mktemp cpuinfo.XXXXXXXXXX)
+# Time in seconds is required to let the benchmark settle down
+SETTLE_TIME="${SETTLE_TIME:-20}"
+# Set the runtime if not set already
+RUNTIME="${RUNTIME:-kata-runtime}"
+
+function main() {
+	# Check tools/commands dependencies
+	cmds=("awk" "docker" "perf")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	check_dockerfiles_images "$IMAGE" "$DOCKERFILE"
+
+	# Start iperf3 server configuration
+	# Set the TMPDIR to an existing tmpfs mount to avoid a 9p unlink error
+	local init_cmds="export TMPDIR=/dev/shm"
+	local server_cmd="$init_cmds && iperf -s"
+	docker run --runtime $RUNTIME --name $SERVER_NAME -d $IMAGE bash -c "$server_cmd"
+	local server_address=$(docker inspect --format "{{.NetworkSettings.IPAddress}}" $SERVER_NAME)
+
+	# Start iperf3 client
+	local client_cmd="$init_cmds && iperf -c $server_address -b $BANDWIDTH -t $TIMEOUT"
+	docker run --runtime $RUNTIME --name $CLIENT_NAME -d $IMAGE bash -c "$client_cmd"
+
+	# This time in seconds is required to let the benchmark settle down
+	sleep "$SETTLE_TIME"
+
+	# Retrieve qemu information
+	PIDS="$(pgrep -d ',' qemu)"
+
+	metrics_json_init
+
+	# Start collecting cpu information
+	sudo perf stat -o $TMP_FILE -e cycles -e instructions -p ${PIDS} sleep ${SAMPLE_PERIOD}
+
+	# Save configuration
+	metrics_json_start_array
+
+	local result=$(cat $TMP_FILE)
+	local instructions_per_cycle="$(echo "$result" | grep -w "insns per cycle" | cut -d '#' -f2 | awk '{print $1}')"
+	local cycles="$(echo "$result" | grep -w "cycles" | awk '{print $1}' | tr ',' ' ' | \
+		awk '{for(i=1;i<=NF;i++) if($i>maxval) maxval=$i;}; END { print maxval;}')"
+	local instructions="$(echo "$result" | grep -w "instructions" | awk '{print $1}' | tr ',' ' ' | \
+		awk '{for(i=1;i<=NF;i++) if($i>maxval) maxval=$i;}; END { print maxval;}')"
+
+	local json="$(cat << EOF
+	{
+		"instructions per cycle": {
+			"Result" : $instructions_per_cycle,
+			"Units"  : "ins per cycle"
+		},
+		"cycles": {
+			"Result" : $cycles,
+			"Units"  : "cycles"
+		},
+		"instructions": {
+			"Result" : $instructions,
+			"Units"  : "instructions"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+	clean_env
+	rm -rf $TMP_FILE
```

Comment:
As requested, can you put the rm of the TMP_FILE into a bash TRAP please, so we don't litter tmp files around if we crash out of the test. Something similar to https://github.com/kata-containers/tests/blob/master/.ci/run_metrics_PR_ci.sh#L74
And to be really safe, you might want to do that as the first command in the script, as we call `mktemp` very early via the env var setup.
thx